### PR TITLE
builder-next: print warning/error for unsupported ImageBuildOptions

### PR DESCRIPTION
**Pull request by AkihiroSuda**
* Prohibit Squash
* Print warning for other unsupported options

Signed-off-by: Akihiro Suda <suda.akihiro@lab.ntt.co.jp>

@tonistiigi @tiborvass

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
@tonistiigi 

Question: can we implement a new executor with `builder.ExecBackend` instead of the current runcexecutor, so that we can easily implement these runtime options?

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Thanks! Left some thoughts / suggestions

File: builder/builder-next/builder.go [hunk 1]
```diff
@@ -317,6 +318,32 @@ func (b *Builder) Build(ctx context.Context, opt backend.BuildConfig) (*builder.
 		exporterAttrs["name"] = strings.Join(opt.Options.Tags, ",")
 	}
 
+	if opt.Options.Squash {
+		// return error rather than print a warning, as squash is often used for hiding secrets
```
[the remark this answers is not in the document]
Comment (AkihiroSuda) [reply builder/builder-next/builder.go]: ping @tonistiigi

[the remark this answers is not in the document]
Comment (tonistiigi) [reply builder/builder-next/builder.go]: Why doesn't squash work? For other options, I don't think we should error on most of them. Maybe only `SecurityOpt` that we probably just don't want anymore. If the removal options are set there is no issue for the user cause buildkit is already guaranteed to not leak containers or partial images. For the cgroup values I think they are optional and warn is ok.

[the remark this answers is not in the document]
Comment (AkihiroSuda) [reply builder/builder-next/builder.go]: > Why doesn't squash work? 

seems a bug in `daemon/image/images.go:SquashImage()`?

[the remark this answers is not in the document]
Comment (AkihiroSuda) [reply builder/builder-next/builder.go]: Looks like squash itself is working but the resulting image in the store is not tagged with the squashed image ID

[the remark this answers is not in the document]
Comment (AkihiroSuda) [reply builder/builder-next/builder.go]: https://github.com/moby/moby/pull/38902

**Comment by AkihiroSuda** [comment]
As `--squash` turned out to be supported (but broken until https://github.com/moby/moby/pull/38902 gets merged), there is no longer reason to return `error`.

For warning messages, probably they should be implemented on CLI side.

**Pull request closed without merging**